src/java.base/share/classes/java/util/Date.java

```diff
@@ -975,10 +970,11 @@ static final long getMillisOf(Date date) {
      * @since   1.2
      * @throws    NullPointerException if {@code anotherDate} is null.
      */
+    @Override
     public int compareTo(Date anotherDate) {
         long thisTime = getMillisOf(this);
         long anotherTime = getMillisOf(anotherDate);
-        return (thisTime<anotherTime ? -1 : (thisTime==anotherTime ? 0 : 1));
+        return Long.compare(thisTime, anotherTime);
```

Comment:
Looks like local variables can also be dropped here as each value is used once.

Reply:
Inlined